cloudpathlib/client.py

```diff
@@ -83,7 +83,9 @@ def _list_dir(
         pass
 
     @abc.abstractmethod
-    def _move_file(self, src: BoundedCloudPath, dst: BoundedCloudPath) -> BoundedCloudPath:
+    def _move_file(
+        self, src: BoundedCloudPath, dst: BoundedCloudPath, remove_src: bool = True
```

Comment:
Personally, I'd find it more intuitive to have separate methods `_move_file` and `_copy_file`, where the base implementation of `_move_file` would be something like:

```python
def _move_file(self, src: BoundedCloudPath, dst: BoundedCloudPath) -> BoundedCloudPath:
    self._copy_file(src, dst)
    if src != dst:
        self._remove(src)
    return dst
```

Reply:
Opened #143